### PR TITLE
Fix \limsup throws exception

### LINKED ISSUE #244: \limsup throws exception
Try to parse the following formula: `\limsup`

It will break the library:
```
Unhandled exception. System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ArgumentException: An item with the same key has already been added. Key: f
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at WpfMath.TexPredefinedFormulaParser.CreateTeXFormulaParser.Parse(SourceSpan source, XElement element) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexPredefinedFormu
laParser.cs:line 175
   at WpfMath.TexPredefinedFormulaParser.ParseFormula(SourceSpan source, XElement formulaElement) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexPredefinedFormulaParser.
cs:line 117
   at WpfMath.TexPredefinedFormulaParser.<>c__DisplayClass10_0.<Parse>b__0(SourceSpan source) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexPredefinedFormulaParser.cs:l
ine 103
   at WpfMath.TexFormulaParser.ProcessEscapeSequence(TexFormula formula, SourceSpan value, Int32& position, Boolean allowClosingDelimiter, Boolean& closedDelimiter, ICom
mandEnvironment environment) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexFormulaParser.cs:line 652
   at WpfMath.TexFormulaParser.Parse(SourceSpan value, Int32& position, Boolean allowClosingDelimiter, String textStyle, ICommandEnvironment environment) in D:\X-Files\P
rojects\wpf-math\src\WpfMath\TexFormulaParser.cs:line 246
   at WpfMath.TexFormulaParser.Parse(String value, String textStyle) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexFormulaParser.cs:line 166
   at WpfMath.TexPredefinedFormulaParser.CreateTeXFormulaParser.Parse(SourceSpan source, XElement element) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexPredefinedFormu
laParser.cs:line 168
   at WpfMath.TexPredefinedFormulaParser.ParseFormula(SourceSpan source, XElement formulaElement) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexPredefinedFormulaParser.
cs:line 117
   at WpfMath.TexPredefinedFormulaParser.<>c__DisplayClass10_0.<Parse>b__0(SourceSpan source) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexPredefinedFormulaParser.cs:l
ine 103
   at WpfMath.TexFormulaParser.ProcessEscapeSequence(TexFormula formula, SourceSpan value, Int32& position, Boolean allowClosingDelimiter, Boolean& closedDelimiter, ICom
mandEnvironment environment) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexFormulaParser.cs:line 652
   at WpfMath.TexFormulaParser.Parse(SourceSpan value, Int32& position, Boolean allowClosingDelimiter, String textStyle, ICommandEnvironment environment) in D:\X-Files\P
rojects\wpf-math\src\WpfMath\TexFormulaParser.cs:line 246
   at WpfMath.TexFormulaParser.Parse(SourceSpan value, String textStyle, ICommandEnvironment environment) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexFormulaParser.cs
:line 177
   at WpfMath.TexFormulaParser.ProcessEscapeSequence(TexFormula formula, SourceSpan value, Int32& position, Boolean allowClosingDelimiter, Boolean& closedDelimiter, ICom
mandEnvironment environment) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexFormulaParser.cs:line 667
   at WpfMath.TexFormulaParser.Parse(SourceSpan value, Int32& position, Boolean allowClosingDelimiter, String textStyle, ICommandEnvironment environment) in D:\X-Files\P
rojects\wpf-math\src\WpfMath\TexFormulaParser.cs:line 246
   at WpfMath.TexFormulaParser.Parse(String value, String textStyle) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexFormulaParser.cs:line 166
   at WpfMath.TexFormulaHelper.AddOperator(String operatorFormula, Boolean useVerticalLimits) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexFormulaHelper.cs:line 158
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at WpfMath.TexPredefinedFormulaParser.MethodInvocationParser.Parse(SourceSpan source, XElement element) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexPredefinedFormu
laParser.cs:line 144
   at WpfMath.TexPredefinedFormulaParser.ParseFormula(SourceSpan source, XElement formulaElement) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexPredefinedFormulaParser.
cs:line 117
   at WpfMath.TexPredefinedFormulaParser.<>c__DisplayClass10_0.<Parse>b__0(SourceSpan source) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexPredefinedFormulaParser.cs:l
ine 103
   at WpfMath.TexFormulaParser.ProcessEscapeSequence(TexFormula formula, SourceSpan value, Int32& position, Boolean allowClosingDelimiter, Boolean& closedDelimiter, ICom
mandEnvironment environment) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexFormulaParser.cs:line 652
   at WpfMath.TexFormulaParser.Parse(SourceSpan value, Int32& position, Boolean allowClosingDelimiter, String textStyle, ICommandEnvironment environment) in D:\X-Files\P
rojects\wpf-math\src\WpfMath\TexFormulaParser.cs:line 246
   at WpfMath.TexFormulaParser.Parse(String value, String textStyle) in D:\X-Files\Projects\wpf-math\src\WpfMath\TexFormulaParser.cs:line 166
   at WpfMath.Controls.FormulaControl.CoerceFormula(DependencyObject d, Object baseValue) in D:\X-Files\Projects\wpf-math\src\WpfMath\Controls\FormulaControl.xaml.cs:lin
e 165
   at System.Windows.DependencyObject.ProcessCoerceValue(DependencyProperty dp, PropertyMetadata metadata, EntryIndex& entryIndex, Int32& targetIndex, EffectiveValueEntr
y& newEntry, EffectiveValueEntry& oldEntry, Object& oldValue, Object baseValue, Object controlValue, CoerceValueCallback coerceValueCallback, Boolean coerceWithDeferredR
eference, Boolean coerceWithCurrentValue, Boolean skipBaseValueChecks)
   at System.Windows.DependencyObject.UpdateEffectiveValue(EntryIndex entryIndex, DependencyProperty dp, PropertyMetadata metadata, EffectiveValueEntry oldEntry, Effecti
veValueEntry& newEntry, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType)
   at System.Windows.DependencyObject.InvalidateProperty(DependencyProperty dp, Boolean preserveCurrentValue)
   at System.Windows.DependencyObject.InvalidateProperty(DependencyProperty dp)
   at System.Windows.Data.BindingExpressionBase.Invalidate(Boolean isASubPropertyChange)
   at System.Windows.Data.BindingExpression.TransferValue(Object newValue, Boolean isASubPropertyChange)
   at System.Windows.Data.BindingExpression.ScheduleTransfer(Boolean isASubPropertyChange)
   at MS.Internal.Data.ClrBindingWorker.NewValueAvailable(Boolean dependencySourcesChanged, Boolean initialValue, Boolean isASubPropertyChange)
   at MS.Internal.Data.PropertyPathWorker.UpdateSourceValueState(Int32 k, ICollectionView collectionView, Object newValue, Boolean isASubPropertyChange)
   at MS.Internal.Data.PropertyPathWorker.OnDependencyPropertyChanged(DependencyObject d, DependencyProperty dp, Boolean isASubPropertyChange)
   at MS.Internal.Data.ClrBindingWorker.OnSourceInvalidation(DependencyObject d, DependencyProperty dp, Boolean isASubPropertyChange)
   at System.Windows.Data.BindingExpression.HandlePropertyInvalidation(DependencyObject d, DependencyPropertyChangedEventArgs args)
   at System.Windows.Data.BindingExpressionBase.OnPropertyInvalidation(DependencyObject d, DependencyPropertyChangedEventArgs args)
   at System.Windows.Data.BindingExpression.OnPropertyInvalidation(DependencyObject d, DependencyPropertyChangedEventArgs args)
   at System.Windows.DependentList.InvalidateDependents(DependencyObject source, DependencyPropertyChangedEventArgs sourceArgs)
   at System.Windows.DependencyObject.NotifyPropertyChange(DependencyPropertyChangedEventArgs args)
   at System.Windows.DependencyObject.UpdateEffectiveValue(EntryIndex entryIndex, DependencyProperty dp, PropertyMetadata metadata, EffectiveValueEntry oldEntry, Effecti
veValueEntry& newEntry, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType)
   at System.Windows.DependencyObject.SetValueCommon(DependencyProperty dp, Object value, PropertyMetadata metadata, Boolean coerceWithDeferredReference, Boolean coerceW
ithCurrentValue, OperationType operationType, Boolean isInternal)
   at System.Windows.DependencyObject.SetCurrentDeferredValue(DependencyProperty dp, DeferredReference deferredReference)
   at System.Windows.Controls.TextBox.OnTextContainerChanged(Object sender, TextContainerChangedEventArgs e)
   at System.Windows.Documents.TextContainer.EndChange(Boolean skipEvents)
   at System.Windows.Documents.TextContainer.System.Windows.Documents.ITextContainer.EndChange(Boolean skipEvents)
   at System.Windows.Documents.TextRangeBase.EndChange(ITextRange thisRange, Boolean disableScroll, Boolean skipEvents)
   at System.Windows.Documents.TextRange.System.Windows.Documents.ITextRange.EndChange(Boolean disableScroll, Boolean skipEvents)
   at System.Windows.Documents.TextRange.ChangeBlock.System.IDisposable.Dispose()
   at System.Windows.Documents.TextEditorTyping.DoTextInput(TextEditor This, String textData, Boolean isInsertKeyToggled, Boolean acceptControlCharacters)
   at System.Windows.Documents.TextEditorTyping.TextInputItem.Do()
   at System.Windows.Documents.TextEditorTyping.ScheduleInput(TextEditor This, InputItem item)
   at System.Windows.Documents.TextEditorTyping.OnTextInput(Object sender, TextCompositionEventArgs e)
   at System.Windows.Controls.Primitives.TextBoxBase.OnTextInput(TextCompositionEventArgs e)
   at System.Windows.UIElement.OnTextInputThunk(Object sender, TextCompositionEventArgs e)
   at System.Windows.Input.TextCompositionEventArgs.InvokeEventHandler(Delegate genericHandler, Object genericTarget)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.TextCompositionManager.UnsafeCompleteComposition(TextComposition composition)
   at System.Windows.Input.TextCompositionManager.PostProcessInput(Object sender, ProcessInputEventArgs e)
   at System.Windows.Input.InputManager.RaiseProcessInputEventHandlers(ProcessInputEventHandler postProcessInput, ProcessInputEventArgs processInputEventArgs)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.TextCompositionManager.UnsafeStartComposition(TextComposition composition)
   at System.Windows.Input.TextCompositionManager.PostProcessInput(Object sender, ProcessInputEventArgs e)
   at System.Windows.Input.InputManager.RaiseProcessInputEventHandlers(ProcessInputEventHandler postProcessInput, ProcessInputEventArgs processInputEventArgs)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   at System.Windows.Interop.HwndKeyboardInputProvider.ProcessTextInputAction(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.OnPreprocessMessage(Object param)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at System.Windows.Interop.HwndSource.OnPreprocessMessageThunk(MSG& msg, Boolean& handled)
   at System.Windows.Interop.ComponentDispatcherThread.RaiseThreadMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.Run()
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run()
   at WpfMath.Example.App.Main()

```

### PR DESCRIPTION
**Pull request by rstm-sf**
Fixes #244 

It may be necessary to full-copy the dictionary, but I think it's unnecessary

### CODE REVIEW

File: src/WpfMath.Tests/TestResults/ParserTests.limsup.approved.txt [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        "End": 16,
+        "Length": 8,
+        "Source": "\\mathrm{lim\\,sup}"
+      }
+    },
+    "LowerLimitAtom": null,
+    "UpperLimitAtom": null,
+    "UseVerticalLimits": true,
+    "Type": "BigOperator",
+    "Source": {
```
Comment (rstm-sf) [inline src/WpfMath.Tests/TestResults/ParserTests.limsup.approved.txt]: But we again lost the original source

Comment (rstm-sf) [reply src/WpfMath.Tests/TestResults/ParserTests.limsup.approved.txt]: >    "Start": 8,
>    "End": 16,
>    "Length": 8,
>    "Source": "\\mathrm{lim\\,sup}"

**Comment by rstm-sf** [comment]
I just tweaked the `ParserTests.limsup.approved.txt`

**Comment by ForNeVeR** [comment]
Thanks for help! This PR is obsoleted by #286 (and I've incorporated your commit there).

**Pull request closed without merging**